### PR TITLE
feat(core): new favorite

### PR DESCRIPTION
**Pull request by EYHN**

### CODE REVIEW

**Comment by EYHN** [comment]
* **#7590** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7590?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7456** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7456?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#7586](https://github.com/toeverything/AFFiNE/pull/7586) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7586?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#7589** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7589?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7588** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7588?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @EYHN and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/common/infra/src/modules/db/services/db.ts [hunk 1]
```diff
@@ -26,7 +38,33 @@ export class DBService extends Service {
...
+    if (userdataDb) {
+      return userdataDb.obj;
+    }
+
+    const newDB = new WorkspaceUserdataDBClient(
+      new YjsDBAdapter(AFFiNE_WORKSPACE_USERDATA_DB_SCHEMA, {
+        getDoc: guid => {
+          const ydoc = new YDoc({
+            // guid format: userdata${userId}${workspaceId}${guid}
+            guid: `userdata$${userId}$${this.workspaceService.workspace.id}$${guid}`,
```
Comment (forehalo) [inline packages/common/infra/src/modules/db/services/db.ts:54]: damn it

File: packages/common/infra/src/orm/core/adapters/yjs/table.ts [hunk 2]
```diff
@@ -67,7 +67,7 @@ export class YjsTableAdapter implements TableAdapter {
         record.set(key, data[key]);
       }
 
-      record.delete(this.deleteFlagKey);
+      record.set(this.deleteFlagKey, false);
```
Comment (forehalo) [inline packages/common/infra/src/orm/core/adapters/yjs/table.ts]: what for?

File: packages/frontend/core/src/components/page-list/operation-cell.tsx [hunk 3]
```diff
@@ -63,7 +69,7 @@ export const PageOperationCell = ({
   const { appSettings } = useAppSettingHelper();
   const { setTrashModal } = useTrashModalHelper(currentWorkspace.docCollection);
   const [openDisableShared, setOpenDisableShared] = useState(false);
-  const favAdapter = useService(FavoriteItemsAdapter);
+  const favAdapter = useService(CompatibleFavoriteItemsAdapter);
   const favourite = useLiveData(favAdapter.isFavorite$(page.id, 'doc'));
```
Comment (forehalo) [inline packages/frontend/core/src/components/page-list/operation-cell.tsx:73]: old typo

File: packages/frontend/core/src/modules/explorer/views/nodes/collection/index.tsx [hunk 4]
```diff
@@ -251,7 +251,7 @@ const ExplorerCollectionNodeChildren = ({
       [docsService]
     )
   );
-  const favourites = useLiveData(favoriteItemsAdapter.favorites$);
+  const favourites = useLiveData(compatibleFavoriteItemsAdapter.favorites$);
```
Comment (forehalo) [inline packages/frontend/core/src/modules/explorer/views/nodes/collection/index.tsx:254]: favorites

**Review by forehalo — APPROVED** [review]

File: packages/frontend/core/src/modules/explorer/views/nodes/collection/index.tsx (hunk 4 above)
Comment (EYHN) [reply packages/frontend/core/src/modules/explorer/views/nodes/collection/index.tsx:254]: CleanShot 2024-07-26 at 15 49 38
too many, i will do that in other PR

File: packages/common/infra/src/orm/core/adapters/yjs/table.ts (hunk 2 above)
Comment (EYHN) [reply packages/common/infra/src/orm/core/adapters/yjs/table.ts]: reverted

File: packages/common/infra/src/modules/db/services/db.ts (hunk 1 above)
Comment (EYHN) [reply packages/common/infra/src/modules/db/services/db.ts:54]: damn it

**Pull request merged**